src/site/content/en/reliable/codelab-background-sync/index.md

```diff
@@ -0,0 +1,212 @@
+- Declares a global error handler, by calling `workbox.routing.setCatchHandler()` to manage failed requests. When requests are for documents, a fallback offline HTML page will be returned.
+
+To test this functionality:
+
+1. Open the demo site in Chrome.
+{% Instruction 'devtools-network', 'ol' %}
+1. In the [Throttling drop-down list](https://developers.google.com/web/tools/chrome-devtools/network/reference#throttling) in DevTools, select **Offline**.
+1. Enter a search query, and click the search button.
+
+The fallback HTML page is shown:
+
+![A screenshot of the custom offline UX in the browser.](offline-ux-custom.png)
+
+## Request notification permission
+
+For simplicity, the offline page at `views/index_offline.html` already contains the code to request notification permissions in a script block at the bottom:
+
+```javascript
+function requestNotificationPermission(event) {
+  event.preventDefault();
+
+  Notification.requestPermission().then(function (result) {
+    showOfflineText(result);
+  });
+}
+```
+
+The code does the following:
+
+- When the user clicks "subscribe to notifications" the `requestNotificationPermission()` function is called, which calls `Notification.requestPermission()`, to show the default browser permission prompt. The promise resolves with the permission picked by the user, which can be either `granted`, `denied` or `default`.
+- Passes the resolved permission to  `showOfflineText()` to show the appropriate text to the user.
+
+## Persist offline queries and retry when back online
+
+Next, implement [backgroundSync](https://developers.google.com/web/tools/workbox/modules/workbox-background-sync) to persist offline queries, so they can be retried when the browser detects that connectivity has returned.
+
+1. Open `public/sw.js` for edit.
+1. Add the following code at the end of the file:
+
+```javascript
+const bgSyncPlugin = new workbox.backgroundSync.Plugin('offlineQueryQueue', {
+  maxRetentionTime: 60,
+  onSync: async ({queue}) => {
+    let entry;
+    while ((entry = await queue.shiftRequest())) {
+      try {
+        const response = await fetch(entry.request);
+        const cache = await caches.open('offline-search-responses');
+        const offlineUrl = entry.request.url + '&notification=true';
+        cache.put(offlineUrl, response);
+        showNotification(offlineUrl);
+      } catch (error) {
+        await this.unshiftRequest(entry);
+        throw error;
+      }
+    }
+  },
+});
+```
+
+The code does the following:
+
+- `workbox.backgroundSync.Plugin`: contains the logic to add failed requests to a queue so they can be retried later. These requests will be persisted in [IndexedDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API).
+- maxRetentionTime indicates the amount of time a request may be retried. In this case we have chosen 60 minutes (after which it will be discarded).
```

Comment:
```suggestion
- `maxRetentionTime` indicates the amount of time a request may be retried. In this case we have chosen 60 minutes (after which it will be discarded).
```